### ISSUE TITLE
Add useful functions

### ISSUE THREAD

**Issue by Gife72** [azymohliad/watchmate#46]
Hi, as per our chat on telegram, I’d like to improve this app with new, useful, functions:
1) distance in case of running/jogging (it’d be done using GPS of Librem 5/other OS and showing into the app how many km/miles
2) sleep tracking (quite easy because is set on heart rate during sleep so it’s possible to figure out the 4 phases (REM, deep sleep,..) and set an intelligent alarm that allows us to wake up feeling sleep-satisfied instead of tired
3) sport parameters (VO2max and Fatmax basically)
VO2max is max volume of oxygen during sport. Formula is = HRmax/HRmin(usually during sleeping) x 15
Fatmax is range of heart rate to have maximum burns of fat. Formula is = between 60% and 80% of VO2max

So, basically, utilise and interpretate heart rate data from Pinetime. What do you think?

**Comment by azymohliad** [comment]
Hi, thanks for the detailed explanations. This is much bigger of a change than I interpreted from your first message. I wouldn't want it to pivot it into a full featured sports tracker app. First of all, I want it to have a scope limited to interfacing with a smart watch. But also, it's just a lot of work that I don't have a bandwidth for. Sorry. I think a more viable approach would be to integrate with [GNOME Health](https://apps.gnome.org/Health/) (see also #30) or maybe [Jogger](https://codeberg.org/baarkerlounger/jogger) or similar apps via a standard interface, so that Watchmate or any other smart watch companion app would be merely a data provider.

Still, for the data coming form the watch alone it might make sense to do some simple processing. I think the first step would be to implement the biometrics recording (heart rate, step count), which would be needed for the integration with health & sports tracker app anyway. Then it would make sense to see how good the data is and experiment with some processing.

To answer to each point:
1. I think GPS distance measurement is out of scope here
2. It might make sense to label sleep phases on the HR plot. The intelligent alarm sounds like a firmware feature to me at first glance, but if the processing is too heavy for it, it might actually be a good fit for a companion app. But I'm not sure how comfortable it would be for sleep tracking and if anybody would use it for that because PineTime heart rate sensor has this annoying green flashing light. 
3. If the integration approach kicks off, then these sounds like a better fit for GNOME Health or similar
